src/utils/transactions.rs

```diff
@@ -112,12 +118,7 @@ pub(super) fn cancellation_parts_hash(safe_address: &Address, nonce: u64) -> [u8
     keccak256(encoded_parts)
 }
 
-pub(super) fn use_legacy_domain_separator(safe_info: Option<SafeInfo>) -> bool {
-    let version = safe_info
-        .as_ref()
-        .and_then(|safe_info| safe_info.version.as_ref().map(|it| Version::parse(it).ok()))
-        .flatten();
-
+pub(super) fn use_legacy_domain_separator(version: Option<Version>) -> bool {
```

Comment:
nitpick: shall we name it `uses_legacy_domain_separator` instead? `use_legacy_domain_separator` seems like it's a "decorator" pattern. Wdyt?

Reply:
Hmm I always read it like asking the method "should I use the legacy domain separator". And your would be more "does this version use the legacy domain separator". No preference from my side.

Reply:
Ah I see your point. No preference either 👍